File: ModularExponent/Python/ModularExponent.py
def modularExponent(base, power, mod):
    
    if mod == 1:
        return 0

    result = 1
    for i in range(power):
        result = (base * result) % mod

    return result

print(modularExponent(4, 8, 9))
print(modularExponent(111, 23, 7))
print(modularExponent(4, 13, 497))

File: ModularExponent/C/ModularExponent.c
#include<stdio.h>

int modularExponent(int base, int power, int mod)
{
	if(mod == 1)
		return 0;

	int result = 1;
	for(int i = 0; i < power; i++)
		result = (base * result) % mod;

	return result;
}

int main()
{
	printf("%d\n", modularExponent(4, 8, 9));
	printf("%d\n", modularExponent(111, 23, 7));
	printf("%d\n", modularExponent(4, 13, 497));

	return 0;
}
